src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ServiceDescriptor.cs

```diff
@@ -1058,8 +1048,6 @@ private string DebuggerToString()
             return debugText;
         }
 
-        private static void ThrowKeyedDescriptor() => throw new InvalidOperationException(SR.KeyedDescriptorMisuse);
-
         private static void ThrowNonKeyedDescriptor() => throw new InvalidOperationException(SR.NonKeyedDescriptorMisuse);
```

Comment:
If we don't throw when accessing on non-leyed property when the descriptor is kedyed, shoud we also remove this, to be consistent?

Reply:
I was wondering about that as well. The original issue was a backwards compat issue with existing APIs so not throwing there makes sense. Throwing for the new APIs I think is OK and desired to help catch issues (as originally intended).

Other than consistency, can you think of a scenario where we would want to return null for the new APIs?

Reply:
Nope it's only for consistency's sake. It's fine to me to throw here otherwise